subsys/bluetooth/host/gatt.c

```diff
@@ -5158,7 +5170,14 @@ void bt_gatt_connected(struct bt_conn *conn)
 
 #if defined(CONFIG_BT_GATT_CLIENT)
 	add_subscriptions(conn);
+#if defined(CONFIG_BT_GATT_AUTO_UPDATE_MTU)
+	int err;
 
+	err = bt_gatt_exchange_mtu(conn, &gatt_exchange_params);
+	if (err) {
+		BT_WARN("MTU Exchange failed (err %d)", err);
+	}
+#endif /* CONFIG_BT_GATT_AUTO_UPDATE_MTU */
 #endif /* CONFIG_BT_GATT_CLIENT */
```

Comment:
- Subscription upkeep and MTU exchange are unrelated. We should keep it separate in code as well.
- It's simpler if enabling automatic MTU exchange depends only on `CONFIG_BT_GATT_AUTO_UPDATE_MTU`. That is what that kconfig's name means to me.
- It's more correct if this fails to compile if `depends on BT_GATT_CLIENT` is removed from `CONFIG_BT_GATT_AUTO_UPDATE_MTU` but `bt_gatt_exchange_mtu` is not defined instead of silently ignoring `CONFIG_BT_GATT_AUTO_UPDATE_MTU`.

```suggestion
#endif /* CONFIG_BT_GATT_CLIENT */

#if defined(CONFIG_BT_GATT_AUTO_UPDATE_MTU)
	int err;

	err = bt_gatt_exchange_mtu(conn, &gatt_exchange_params);
	if (err) {
		BT_WARN("MTU Exchange failed (err %d)", err);
	}
#endif /* CONFIG_BT_GATT_AUTO_UPDATE_MTU */
```